L1Trigger/L1TMuonCPPF/src/RecHitProcessor.cc

```diff
@@ -0,0 +1,462 @@
+	    local_phi = global_phi-15.;
+	    EMTFsector1 = 1;
+	    EMTFsector2 = 0;
+	  }
+	  if ((global_phi > 53.) && (global_phi <= 75.)) {
+	    local_phi = global_phi-15.;
+	    EMTFsector1 = 1;
+	    EMTFsector2 = 2;
+	  }
+	  //sector 2 
+	  if ((global_phi > 75.) && (global_phi <= 76.3)) {
+	    local_phi = global_phi-15.;
+	    EMTFsector1 = 1;
+	    EMTFsector2 = 2;
+	  }
+	  if ((global_phi > 76.3) && (global_phi <= 113.)) {
+	    local_phi = global_phi-75.;
+	    EMTFsector1 = 2;
+	    EMTFsector2 = 0;
+	  }
+	  if ((global_phi > 113.) && (global_phi <= 135.)) {
+	    local_phi = global_phi-75.;
+	    EMTFsector1 = 2;
+	    EMTFsector2 = 3;
+	  }
+	  //sector 3
+	  //less than 180
+	  if ((global_phi > 135.) && (global_phi <= 136.3)) {
+	    local_phi = global_phi-75.;
+	    EMTFsector1 = 2;
+	    EMTFsector2 = 3;
+	  }
+	  if ((global_phi > 136.3) && (global_phi <= 173.)) {
+	    local_phi = global_phi-135.;
+	    EMTFsector1 = 3;
+	    EMTFsector2 = 0;
+	  }
+	  if ((global_phi > 173.) && (global_phi <= 180.)) {
+	    local_phi = global_phi-135.;
+	    EMTFsector1 = 3;
+	    EMTFsector2 = 4;
+	  }
+	  //Greater than -180
+	  if ((global_phi < -165.) && (global_phi >= -180.)) {
+	    local_phi = global_phi+225.;
+	    EMTFsector1 = 3;
+	    EMTFsector2 = 4;
+	  }
+	  //Fourth sector
+	  if ((global_phi > -165.) && (global_phi <= -163.7)) {
+	    local_phi = global_phi+225.;
+	    EMTFsector1 = 3;
+	    EMTFsector2 = 4;
+	  }
+	  if ((global_phi > -163.7) && (global_phi <= -127.)) {
+	    local_phi = global_phi+165.;
+	    EMTFsector1 = 4;
+	    EMTFsector2 = 0;
+	  }
+	  if ((global_phi > -127.) && (global_phi <= -105.)) {
+	    local_phi = global_phi+165.;
+	    EMTFsector1 = 4;
+	    EMTFsector2 = 5;
+	  }
+	  //fifth sector
+	  if ((global_phi > -105.) && (global_phi <= -103.7)) {
+	    local_phi = global_phi+165.;
+	    EMTFsector1 = 4;
+	    EMTFsector2 = 5;
+	  }
+	  if ((global_phi > -103.7) && (global_phi <= -67.)) {
+	    local_phi = global_phi+105.;
+	    EMTFsector1 = 5;
+	    EMTFsector2 = 0;
+	  }
+	  if ((global_phi > -67.) && (global_phi <= -45.)) {
+	    local_phi = global_phi+105.;
+	    EMTFsector1 = 5;
+	    EMTFsector2 = 6;
+	  } 
+	  //sixth sector
+	  if ((global_phi > -45.) && (global_phi <= -43.7)) {
+	    local_phi = global_phi+105.;
+	    EMTFsector1 = 5;
+	    EMTFsector2 = 6;
+	  } 
+	  if ((global_phi > -43.7) && (global_phi <= -7.)) {
+	    local_phi = global_phi+45.;
+	    EMTFsector1 = 6;
+	    EMTFsector2 = 0;
+	  }
+	  if ((global_phi > -7.) && (global_phi <= 15.)) {
+	    local_phi = global_phi+45.;
+	    EMTFsector1 = 6;
+	    EMTFsector2 = 1;
+	  } 
+	  
+	  int int_phi = int((local_phi + 22.0 )*15. + .5); 
+	  
+	  double EMTFLink1 = 0.;
+	  double EMTFLink2 = 0.;
+          double lb = 0.;
+	  double halfchannel = 0.;
+	  
+	  //Invalid hit
+	  if (isValid == 0) int_phi = 2047;	 
+	  //Right integers range
+	  assert(0 <= int_phi && int_phi < 1250);
```

Comment:
Maybe some error message here?

Reply:
No error message due to the range of this variable.